File: Assignment48/program5.java
import java.util.*;

class program5 {
    public static void main(String A[]){
        Scanner sobj = new Scanner(System.in);

        int PurchaseAmount = 0;
        double discount = 0;
        double discountedAmount = 0;
        double finalAmount = 0;

        String membership = "";

        System.out.println("Enter the purchase amount : ");
        PurchaseAmount = sobj.nextInt();

        System.out.println("Enter membership type (Premium/Regular) : ");
        membership = sobj.next();

        if(PurchaseAmount <= 0) {
            System.out.println("Invalid amount");
        }
        else if(!membership.equals("Premium") && !membership.equals("Regular")) {
            System.out.println("Invalid membership type");
        }
        else{
            if(PurchaseAmount > 5000) {
                discount = PurchaseAmount * 0.20;
            }
            else if(PurchaseAmount > 2000) {
                discount = PurchaseAmount * 0.10;
            }
            else {
                discount = 0;
            }
            discountedAmount = PurchaseAmount - discount;

            if(membership.equalsIgnoreCase("Premium")) {
                double premiumDiscount = discountedAmount * 0.05;
                discount = discount + premiumDiscount;
                finalAmount = discountedAmount - premiumDiscount;
            }
            else {
                finalAmount = discountedAmount;
            }

            System.out.println("Original Amount: ₹" + PurchaseAmount);
            System.out.println("Total Discount: ₹" + discount);
            System.out.println("Final Payable Amount: ₹" + finalAmount);
        }
    }
}

File: Assignment49/program5.cpp
#include<iostream>
using namespace std;

int main(){

    int income = 0;
    double tax = 0;

    cout << "Enter annual income : ";
    cin >> income;

    if(income < 0){
        cout << "Invalid Input" << endl;
        return 0;
    }

    if(income <= 250000){
        tax = 0;
    }
    else if(income <= 500000){
        tax = (income - 250000) * 0.05;
    }
    else if(income <= 1000000){
        tax = (250000 * 0.05);
        tax += (income - 500000) * 0.20;
    }
    else{
        tax = (250000 * 0.05);
        tax += (500000 * 0.20);
        tax += (income - 1000000) * 0.30;
    }

    cout << "Annual Income: ₹" << income << endl;
    cout << "Total Tax Payable: ₹" << tax << endl;

    return 0;
}